x/nft/msg.go

```diff
@@ -0,0 +1,56 @@
+package nft
+
+import (
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+)
+
+var _ weave.Msg = (*AddApprovalMsg)(nil)
+var _ weave.Msg = (*RemoveApprovalMsg)(nil)
+
+const (
+	pathAddApproval    = "nft/add-approval"
+	pathRemoveApproval = "nft/remove-approval"
+)
+
+// Path returns the routing path for this message
+func (*AddApprovalMsg) Path() string {
+	return pathAddApproval
+}
+
+// Path returns the routing path for this message
+func (*RemoveApprovalMsg) Path() string {
+	return pathRemoveApproval
+}
+
+func (m *AddApprovalMsg) Validate() error {
+	if err := weave.Address(m.Address).Validate(); err != nil {
+		return err
+	}
+	if m.Action == "" {
+		return errors.ErrInternal("action must not be empty")
+	}
+	//TODO: Figure out if we care, as we have a default in ops now that adds infinite count
+	if m.Options == nil {
```

Comment:
pp: I would allow nil in the message to not enforce sending them. alternatively you could use `[(gogoproto.nullable) = false];` in protobuf

Reply:
I would require options, I think it's way more explicit. We can default to something in issue/add/remove messages.